<?xml version="1.0"?>
<launch>
  <arg name="tool_name" default="tool"/> <!-- Name of the tool we are going to use -->
  
  <!-- Parameters to pass to the ROS node -->
  <arg name="joint_position_topic" default="/iiwa/state/JointPosition"/>
  <arg name="cartesian_position_topic" default="/iiwa/state/CartesianPose"/>
  <arg name="command_cartesian_position_topic" default="/iiwa/command/JointPosition"/>
  <arg name="command_joint_position_topic" default="/iiwa/command/CartesianPose"/>
  <arg name="use_cartesian_command" default="true"/>
  <arg name="ros_rate" default="0.1"/>
  
  <param name="/iiwa/publishJointStates" type="boolean" value="true" /> <!--publish joint states (avoid joint_state_controller)-->
  <param name="/iiwa/toolName" type="string" value="$(arg tool_name)" /> <!--Sunrise tool-->

  <!-- Here we call the ROS node we have written, with the parameters on top   -->
  <node ns="iiwa" name="CommandRobot" pkg="iiwa" type="command_robot_sim" respawn="false" output="screen">
    <param name="joint_position_topic" value="$(arg joint_position_topic)"/>
    <param name="cartesian_position_topic" value="$(arg cartesian_position_topic)"/>
    <param name="command_cartesian_position_topic" value="$(arg command_cartesian_position_topic)"/>
    <param name="command_joint_position_topic" value="$(arg command_joint_position_topic)"/>
    <param name="use_cartesian_command" value="$(arg use_cartesian_command)"/>
    <param name="ros_rate" value="$(arg ros_rate)"/>
  </node>
  

</launch>